machine-learning/ml/pipelines/binary.py

```diff
@@ -94,47 +154,18 @@ def _run_single_model(self, model_def, x, y, x_train, x_test, y_train, y_test):
             search = GridSearchCV(model, param_dist, cv=StratifiedKFold(n_splits=N_CV_SEARCH, shuffle=True), iid=False, n_jobs=-1)
 
         # Train and test the model
-        test_scores = self._evaluate_model(search, x_train, x_test, y_train, y_test)
+        test_scores = _evaluate_model(search, x_train, x_tests, y_train, y_tests)
 
         # Run cross validation on whole dataset and safe production ready model
-        super_model = self._build_production_model(model_def, search.best_params_, x, y)
+        super_model = _build_production_model(model_def, search.best_params_, x, y)
 
         # return the scores and the best estimator
         return test_scores, super_model
 
 
-    def _evaluate_model(self, search, x_train, x_test, y_train, y_test):
-        log("Test search started at %s\n" % now())
-        search.fit(x_train, y_train)
-        log(format_best_parameters(search))
-        best_estimator = search.best_estimator_
-
-        # Predict unseen results
-        y_pred = best_estimator.predict(x_test)
-
-        test_scores = {'accuracy': accuracy_score(y_test, y_pred), 'precision': precision_score(y_test, y_pred),
-                        'recall': recall_score(y_test, y_pred), 'tn': confusion_matrix(y_test, y_pred).ravel()[0],
-                        'fp': confusion_matrix(y_test, y_pred).ravel()[1],
-                        'fn': confusion_matrix(y_test, y_pred).ravel()[2],
-                        'tp': confusion_matrix(y_test, y_pred).ravel()[3]}
-
-        return test_scores
-
-    def _build_production_model(self, model_def, best_params, x, y):
-        log("Production model build started at %s\n" % now())
-
-        super_model = model_def.model(best_params)
-        super_model.fit(x, y)
-
-        return super_model
-
-
 class DeepLearningBinaryClassificationPipeline(BinaryClassificationPipeline):
```

Comment:
I think we can delete everything related to deep learning? Or are we going to try some NN?

Reply:
I don't think, we have the time for NN or other deep learning algorithms.
I will remove it in another PR though, to keep this one more concise.

Reply:
https://github.com/refactoring-ai/predicting-refactoring-ml/issues/201